### PR TITLE
Update release instructions

### PR DESCRIPTION
**Pull request by edenhill**
@jliunyu Read the release instructions in its entirety and let me know if anything is unclear.

### CODE REVIEW

File: tools/RELEASE.md [hunk 1]
```diff
@@ -24,163 +24,318 @@ The following guide uses `v0.11.4rc1` as version for a release candidate,
...
+ * Create a release candidate branch.
+ * Update versions, dependencies (librdkafka), etc.
+ * Create test tag to trigger CI builds, test.
+ * Work out any release build issues.
+ * librdkafka is released, update librdkafka version.
+ * Create release candidate tag, wait for CI builds, deploy to test-PyPi.
+ * Create PR, get it reviewed.
+ * When it is time for final release, merge the PR.
+ * On master, create release tag, wait for CI builds, deploy to PyPi.
+ * Announce release.
```
Comment (mhowlett) [inline tools/RELEASE.md:43]: something that was missed last time was updating the API docs to the confluent website, which resulted in some issues. i don't think that's included in this set of instructions?

Comment (edenhill) [reply tools/RELEASE.md:43]: Good point!

**Review by mhowlett — APPROVED** [review]

**Review by jliunyu — APPROVED** [review]
Thanks a lot for updating the release instructions!

File: tools/RELEASE.md [hunk 2]
```diff
@@ -24,163 +24,319 @@ The following guide uses `v0.11.4rc1` as version for a release candidate,
 replace as necessary with your version or remove `rc..` suffix for the
 final release.
 
+To give you an idea what this process looks like we have this fabricated
+time line:
+
+ * Time to prepare for a release.
+ * A librdkafka release candidate is available.
+ * Check if any open PRs needs merging, do that.
```
Comment (jliunyu) [inline tools/RELEASE.md:32]: I found that there are 16 open pull requests https://github.com/confluentinc/confluent-kafka-python/pulls, is there a good way to distinguish which PRs need merging?

**Pull request merged**